### PR TITLE
Refactor 'Tag To Category' Metabox into new flow

### LINKED ISSUE #41: Refactor 'Tag To Category' Metabox into new code Architecture

### PR DESCRIPTION
**Pull request by anandB000**
Fix #41

### CODE REVIEW

File: include/Metabox/Posts/Tagtocategory.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
+<?php
```
Comment (sudar) [inline include/Metabox/Posts/Tagtocategory.php:1]: The filename should be `TagToCategory`. Class name should be `BM_Metabox_Posts_TagToCategory`

File: include/Metabox/Posts/Tagtocategory.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+/**
+ * Bulk Move Posts Tag To Category Metabox.
+ *
+ * @since 2.0.0
+ */
+class BM_Metabox_Posts_Tagtocategory extends BM_Metabox_Base {
+
+    protected function initialize() {
+        $this->meta_box_slug         = 'bm-posts-tag-to-category';
+        $this->messages['box_label'] = __( 'Move Posts Tag To Category', 'bulk-move' );
```
Comment (sudar) [inline include/Metabox/Posts/Tagtocategory.php]: Change label to `Move Posts from Tag to Category`

File: include/Metabox/Posts/Tagtocategory.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        $this->action                = 'move_tagtocategory';
```
Comment (sudar) [inline include/Metabox/Posts/Tagtocategory.php]: This should be `move_tag_to_category`

File: include/Metabox/Posts/Tagtocategory.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                            'orderby'          => 'NAME',
+                            'hide_empty'       => false,
+                        ) );
+                        ?>
+                    </td>
+                </tr>
+
+            </table>
+            <p>
+                <?php _e( 'If the post contains other tags, then', 'bulk-move' ); ?>
```
Comment (sudar) [inline include/Metabox/Posts/Tagtocategory.php]: This should be `If the post contains other categories`

File: include/Metabox/Posts/Tagtocategory.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+        foreach ( $posts as $post ) {
+            $current_tags = wp_get_post_tags( $post->ID, array( 'fields' => 'ids' ) );
+            $current_tags = array_diff( $current_tags, array( $options['tag'] ) );
+
+            $current_cats = wp_get_post_categories( $post->ID );
+
+            if ( $options['overwrite'] ) {
+                // Override is set, so remove all common tags.
+                $current_tags = array();
```
Comment (sudar) [inline include/Metabox/Posts/Tagtocategory.php]: If overwrite is set, then the categories should be reset, not tags.

File: include/BulkMove.php [hunk 6]
```diff
@@ -263,6 +263,7 @@ private function get_post_admin_page() {
 		// TODO: Add other metaboxes.
 		$posts_page->add_metabox( new BM_Metabox_Posts_Category() );
 		$posts_page->add_metabox( new BM_Metabox_Posts_Tag() );
+        $posts_page->add_metabox( new BM_Metabox_Posts_Tagtocategory() );
```
Comment (sudar) [inline include/BulkMove.php]: Alignment is wrong from this line also.

**Review by sudar — CHANGES_REQUESTED** [review]

File: include/Metabox/Posts/Tagtocategory.php (hunk 1 above)
Comment (anandB000) [reply include/Metabox/Posts/Tagtocategory.php:1]: Fixed

File: include/Metabox/Posts/Tagtocategory.php (hunk 2 above)
Comment (anandB000) [reply include/Metabox/Posts/Tagtocategory.php]: Fixed

File: include/Metabox/Posts/Tagtocategory.php (hunk 3 above)
Comment (anandB000) [reply include/Metabox/Posts/Tagtocategory.php]: Fixed

**Review by sudar — APPROVED** [review]

**Pull request merged**